lib/Conversion/PipelineToHW/PipelineToHW.cpp

```diff
@@ -108,16 +155,30 @@ class PipelineLowering {
     if (this->clockGateRegs) {
       // Create the top-level clock gate.
       notStalledClockGate = builder.create<seq::ClockGateOp>(
-          loc, clock, stageValidAndNotStalled, /*test_enable=*/Value());
+          loc, args.clock, stageValidAndNotStalled, /*test_enable=*/Value());
     }
 
     for (auto it : llvm::enumerate(stageTerminator.getRegisters())) {
-      assert(clock && "clock not set");
-      assert(reset && "reset not set");
       auto regIdx = it.index();
       auto regIn = it.value();
-      auto regName = builder.getStringAttr(stageRegPrefix.strref() + "_reg" +
-                                           std::to_string(regIdx));
+
+      // Register naming: If the source value has an `sv.namehint` attribute
+      // attached, use that as a register prefix.
+      StringAttr regName;
+      if (auto *definingOp = regIn.getDefiningOp()) {
```

Comment:
plz ditch the curlies as per llvm style guide.

Reply:
I'd say this is a case where the style guide explicitly says that there **should** be a brace - this does, in my opinion, **not** constitute a (quote) "_simple_" if statement.